plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -99,4 +108,353 @@ public void testPropertiesTable()
+                            ('partition', 'row(_varchar varchar, _date date)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', ''),
+                            ('data', 'row(_bigint row(min bigint, max bigint, null_count bigint))', '', '')
+                            """);
+
+            assertQuery(
+                    "SELECT partition._varchar, partition._date FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('b', DATE '2019-09-09'),
+                            ('c', DATE '2019-09-09'),
+                            ('d', DATE '2019-09-09'),
+                            ('e', DATE '2019-09-10'),
+                            ('f', DATE '2019-09-10'),
+                            (null, DATE '2019-09-10'),
+                            ('g', null),
+                            (null, null)
+                            """);
+
+            assertThat(query("SELECT CAST(data._bigint AS ROW(BIGINT, BIGINT, BIGINT)) FROM \"" + tableName + "$partitions\""))
+                    .matches("""
+                            VALUES 
+                            ROW(ROW(BIGINT '3', BIGINT '3', BIGINT '0')),
+                            ROW(ROW(BIGINT '1', BIGINT '1', BIGINT '0')),
+                            ROW(ROW(BIGINT '6', BIGINT '8', BIGINT '0')),
+                            ROW(ROW(BIGINT '7', BIGINT '7', BIGINT '0')),
+                            ROW(ROW(BIGINT '4', BIGINT '4', BIGINT '0')),
+                            ROW(ROW(BIGINT '6', BIGINT '6', BIGINT '0')),
+                            ROW(ROW(BIGINT '4', BIGINT '50', BIGINT '0')),
+                            ROW(ROW(BIGINT '2', BIGINT '2', BIGINT '0'))
+                            """);
+        }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+    
+    private void testPartitionsTableDataColumn(String type, String columnName, String insertIntoValues, int insertIntoValuesCount, String expectedDataColumn)
+    {
+        String tableName = "test_partitions_table_data_column_" + columnName + "_" + randomNameSuffix();
+        try {
+            assertUpdate("CREATE TABLE " + tableName + "(" + columnName + " " + type + ", _partition VARCHAR) WITH (partitioned_by = ARRAY['_partition'])");
+            assertUpdate("INSERT INTO " + tableName + " " + insertIntoValues, insertIntoValuesCount);
+            assertThat(query("SELECT CAST(data." + columnName + " AS ROW(" + type + "," + type + ", BIGINT)) FROM \"" + tableName + "$partitions\"")).matches(expectedDataColumn);
+        }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableDataColumn()
+    {
+        testPartitionsTableDataColumn("BOOLEAN", "_boolean", "VALUES (true, 'a'), (false, 'a'), (false, 'b'), (false, 'b')", 4, """
```

Comment:
Please refactor the helper method to avoid passing redundant non-partition column `a` and `b`.

Reply:
Do you have recommendation on how not to pass in `a` and `b`, should I stick to one type of partition value? 

Also `a` and `b` are values in the partition column.